lib/BridgedRoom.js

```diff
@@ -0,0 +1,123 @@
+"use strict";
+
+function BridgedRoom(bridge, gitter, gitterRealtime, matrixRoomModel, gitterRoomModel) {
+  this._bridge = bridge;
+  this._gitter = gitter;
+  this._gitterRealtime = gitterRealtime;
+  this._matrixRoomModel = matrixRoomModel;
+  this._gitterRoomModel = gitterRoomModel;
+}
+
+BridgedRoom.prototype.matrixRoomId = function() {
+  return this._matrixRoomModel.getId();
+};
+
+BridgedRoom.prototype.gitterRoomName = function() {
+  return this._gitterRoomModel.getId();
+};
+
+BridgedRoom.prototype.joinAndStart = function() {
+  // We have to find out our own gitter user ID so we can ignore reflections of
+  // messages we sent
+  return this._bridge.getMyGitterUserId().then((gitterUserId) => {
+    this._gitterUserId = gitterUserId;
+
+    return this._gitter.rooms.join(this.gitterRoomName()).then((room) => {
+      this._gitterRoom = room;
+
+      var events = room.streaming().chatMessages();
+
+      events.on('chatMessages', (message) => {
+        if (!message.model.fromUser || message.model.fromUser.id == gitterUserId) {
+          // Ignore a reflection of my own messages
+          return;
+        }
+
+        if (message.operation !== 'create' && message.operation !== 'update') {
+          return;
```

Comment:
Probably just historical. The `onGitterMessage` code used to be inline in the method; it was easier to `return` if the type wasn't `create` and then perform the logic. But then I moved the logic, and discovered `update` operations.

Have now swapped the logic; 6ddefc7